cloudferry/lib/os/migrate/glance.py

```diff
@@ -140,32 +133,30 @@ class UploadImage(BaseImageMigrationTask):
 
     def migrate(self, dst_object, *args, **kwargs):
         if self.src_object.status == 'deleted':
-            return [True]
+            return dict(need_restore_deleted=True)
         image_id = dst_object.object_id.id
         try:
             image_data = _FileLikeProxy(self.src_glance.images.data(image_id))
             self.dst_glance.images.update(image_id, data=image_data)
-            return [False]
+            return dict(need_restore_deleted=False)
         except glance_exc.HTTPNotFound:
-            return [True]
+            return dict(need_restore_deleted=True)
 
 
 class UploadDeletedImage(BaseImageMigrationTask):
-    default_provides = ['status']
-
     def migrate(self, dst_object, need_restore_deleted, *args, **kwargs):
         if not need_restore_deleted:
-            return [True]
+            return
         dst_image_id = dst_object.object_id.id
         with model.Session() as session:
             boot_disk_infos = self._get_boot_disk_locations(session)
 
         for boot_disk_info in boot_disk_infos:
             if self.upload_server_image(boot_disk_info, dst_image_id):
-                return [True]
+                return
         LOG.warning('Unable to restore image %s: no servers found',
                     dst_image_id)
-        return [False]
+        raise RuntimeError('Unable to restore deleted image: no servers found')
```

Comment:
It will fail image migration and don't proceed with migration that depend on them. When migration will complete, exception will be raised that will include all exception raised in tasks based on which we can report what objects we failed to migrate and why.